### PR TITLE
Add GUI with LVGL 

### PR DESCRIPTION
**Pull request by MohammadAbbasi76**
In this pull request, we've integrated a graphical user interface (GUI) using the NXP GUI Guider. We've meticulously crafted and designed all necessary elements within the NXP GUI Guider environment. Subsequently, we've utilized the "generated" and "custom" folders within the NXP GUI Guider.

To streamline the process, we've included the usage of "Print_needed_Cmake.sh" for generating the necessary CMake files.

Additionally, we've incorporated LVGL 8.3.10 and ESP32 IDF 5.2 into our development stack to ensure compatibility and optimal performance.

### CODE REVIEW

File: components/SpotifyInterface/include/SpotifyTypedef.h [hunk 1]
```diff
@@ -18,7 +18,7 @@ extern "C"
 #include "mdns.h"
 #include "freertos/queue.h"
 #include "esp_psram.h"
-#include "SpotifyWebAppInfo.h"
+#include "SpotifyWebAppInfo_sample.h"
```
Comment (Behn4m) [inline components/SpotifyInterface/include/SpotifyTypedef.h]: this is sample, dont fill it with real info

File: components/SpotifyInterface/include/SpotifyWebAppInfo_sample.h
Comment (Behn4m) [inline components/SpotifyInterface/include/SpotifyWebAppInfo_sample.h:1]: should not expose tour client info

File: components/GUI/custom/custom.c
Comment (Behn4m) [inline components/GUI/custom/custom.c:1]: what this source does?

File: components/GUI/generated/events_init.c
Comment (Behn4m) [inline components/GUI/generated/events_init.c:1]: a new line is needed after each function

**Review by Behn4m — APPROVED** [review]

File: components/SpotifyInterface/include/SpotifyTypedef.h (hunk 1 above)
Comment (MohammadAbbasi76) [reply components/SpotifyInterface/include/SpotifyTypedef.h]: we can fix it

File: components/SpotifyInterface/include/SpotifyWebAppInfo_sample.h
Comment (MohammadAbbasi76) [reply components/SpotifyInterface/include/SpotifyWebAppInfo_sample.h:1]: ok

File: components/GUI/custom/custom.c
Comment (MohammadAbbasi76) [reply components/GUI/custom/custom.c:1]: actually ,NXP GUI make this file , but we don't use it , i will remove it

File: components/GUI/generated/events_init.c
Comment (MohammadAbbasi76) [reply components/GUI/generated/events_init.c:1]: Add comment and new line

**Pull request merged**